ultravox/data/datasets.py

```diff
@@ -76,6 +76,12 @@ class DataCollatorForSeq2SeqWithAudio(transformers.DataCollatorForSeq2Seq):
 
     def __call__(self, features, *args, **kwargs):
         audio_values = [f.pop("audio_values", None) for f in features]
+        for i, audio in enumerate(audio_values):
+            audio_values[i] = torch.tensor(audio)
+
+        for i, feature in enumerate(features):
+            features[i]["input_ids"] = torch.tensor(feature["input_ids"])
+
```

Comment:
Can you explain why the changes in this method is needed for batched inference?

Reply:
In training we only use right padding. If `left` padding is applied, all the token positions are gonna move and that includes where the audio is located which we assume is indicated by `audio_token_start_idx`. As such, we need to update this variable to match the reality.